src/node_file.cc

```diff
@@ -187,12 +291,10 @@ void AfterNoArgs(uv_fs_t* req) {
 void AfterStat(uv_fs_t* req) {
   FSReqWrap* req_wrap = static_cast<FSReqWrap*>(req->data);
   FSReqAfterScope after(req_wrap, req);
-  Environment* env = req_wrap->env();
 
   if (after.Proceed()) {
-    FillStatsArray(env->fs_stats_field_array(),
-                   static_cast<const uv_stat_t*>(req->ptr));
-    req_wrap->Resolve(Undefined(req_wrap->env()->isolate()));
+    req_wrap->FillStatsArray(static_cast<const uv_stat_t*>(req->ptr));
```

Comment:
sidenote, using `&req->statbuf` will let you get rid of the cast + remove one layer of indirection

Reply:
Good call